pybamm/solvers/idaklu_solver.py

```diff
@@ -675,3 +676,41 @@ def _integrate(self, model, t_eval, inputs_dict=None):
             return newsol
         else:
             raise pybamm.SolverError("idaklu solver failed")
+
+    def jaxify(
+        self,
+        model,
+        t_eval,
+        *,
+        output_variables=None,
+        inputs=None,
+        calculate_sensitivities=True,
+    ):
+        """JAXify the solver object
+
+        Creates a JAX expression representing the IDAKLU-wrapped solver
+        object.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model to be solved
+        t_eval : numeric type, optional
+            The times at which to compute the solution. If None, the times in the model
+            are used.
+        output_variables : list of str, optional
+            The variables to be returned. If None, the variables in the model are used.
+        inputs : dict, optional
```

Comment:
Yes, while it is true that we don't need `inputs` (or `t_eval`) until evaluation, we do still need knowledge of them in order to setup/jaxify the interface.

I've taken another look at this and I can extract the names of the `inputs` from the model definition (removing the need to include `inputs` in this call). But we also rely on knowing the span of `t_eval` when setting up the idaklu-jax interface. This is because calls to some functions (including `grad`) need to be made over `vmap` (or the user may simply evaluate the expression at only one time point). These details are abstracted away from the user wherever possible in our interface, but we get around this by caching a solve over the predefined `t_span` on a first run.

Hopefully that makes sense. I've removed `inputs` from the call as superfluous, but retained `t_eval` and made a comment in the jupyter notebook to signify that this defines the time-span scope of the solver.